geode-redis/src/main/java/org/apache/geode/redis/internal/data/RedisString.java

```diff
@@ -615,7 +615,7 @@ public int setbit(
    */
 
   @Override
-  public void toData(DataOutput out, SerializationContext context) throws IOException {
+  public synchronized void toData(DataOutput out, SerializationContext context) throws IOException {
```

Comment:
I don't think it's just toData in for RedisString, I believe other methods need to be synchronized as well (like RedisHash and RedisSet do).

Reply:
Also, it would be nice to have a failing test for this if possible?

Reply:
- There are a couple of confirmSerializationIsStable tests and testSerializables. Is there any specific feature that you - like me to check in a test?
- Also, in this commit, I was just primarily undoing the removal of the synchronized blocks that I missed in the previous commit.

Reply:
I will create another ticket to figure out which other methods are not synchronized and fix those. This commit is just to undo the synchronizied commits I missed

Reply:
> * There are a couple of confirmSerializationIsStable tests and testSerializables. Is there any specific feature that you - like me to check in a test?

- added tests

Reply:
Ok. I will approve this PR with the understanding that there will be another fix to synchronize the rest of the memthods in RedisString.

When I was talking about tests, I meant tests that show that we actually need synchronization. Like a test that shows data corruption if we don't have synchronization. I'm not sure there is much value in programatically testing that the method has a synchronization modifier.